src/procmon.cpp

```diff
@@ -27,7 +27,12 @@ static inline int pidfd_open(int pid, unsigned int flags)
 
 ProcMon::ProcMon(const std::string& pid)
 {
-  setup(parse_pid(pid));
+  std::string err;
+  auto maybe_pid = parse_pid(pid, err);
+  if (!maybe_pid.has_value()) {
+    throw std::runtime_error("Failed to parse pid: " + err);
```

Comment:
Looks like in main() it's just a catch + exit anyways: https://github.com/bpftrace/bpftrace/blob/cf404154a06709efcd98171ed5700e3103555136/src/main.cpp#L787-L794

So I think LOG(FATAL) could be appropriate here. IMHO I'd just remove the `std::string&` overload and force the caller to parse the pid.

Reply:
Just an FYI: I tried to add a bunch of `LOG(FATAL)`s to replace all the throws in this file and I ran into two issues:
- checking error behavior in unit tests isn't possible if we call `LOG(FATAL)`
- If I remove the `try/catch` in main then the error output is strange and I couldn't match against it properly in the runtime test (this one probably just needs investigation).

But yeah, I can remove the overload - that will help.

Reply:
Makes sense